modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/MultiTenantQueueTestCase.java

```diff
@@ -0,0 +1,145 @@
+        int runTime = 20;
+        int expectedMessageCount = 200;
+
+        // Start receiving clients (tenant1, tenant2)
+        AndesClient tenant1ReceivingClient = new AndesClient("receive", "127.0.0.1:5672", "queue:topictenant1.com/multitenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(expectedMessageCount),
+                "1", "listener=true,ackMode=1,delayBetweenMsg=0,stopAfter=" + expectedMessageCount, "",
+                "topictenantuser1!topictenant1.com", "topictenantuser1");
+        tenant1ReceivingClient.startWorking();
+
+        AndesClient tenant2ReceivingClient = new AndesClient("receive", "127.0.0.1:5672", "queue:topictenant2.com/multitenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(expectedMessageCount),
+                "1", "listener=true,ackMode=1,delayBetweenMsg=0,stopAfter=" + expectedMessageCount, "",
+                "topictenantuser1!topictenant2.com", "topictenantuser1");
+        tenant2ReceivingClient.startWorking();
+
+        // Start sending clients (tenant1, tenant2)
+
+        AndesClient tenant1SendingClient = new AndesClient("send", "127.0.0.1:5672", "queue:topictenant1.com/multitenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(sendMessageCount1), "1",
+                "ackMode=1,delayBetweenMsg=0,stopAfter=" + sendMessageCount1, "",
+                "topictenantuser1!topictenant1.com", "topictenantuser1");
+
+        tenant1SendingClient.startWorking();
+
+        AndesClient tenant2SendingClient = new AndesClient("send", "127.0.0.1:5672", "queue:topictenant2.com/multitenantTopic",
+                "100", "false", Integer.toString(runTime), Integer.toString(sendMessageCount2), "1",
+                "ackMode=1,delayBetweenMsg=0,stopAfter=" + sendMessageCount2, "",
+                "topictenantuser1!topictenant2.com", "topictenantuser1");
+        tenant2SendingClient.startWorking();
+
+        AndesClientUtils.waitUntilMessagesAreReceived(tenant1ReceivingClient,
+                expectedMessageCount, runTime);
+        AndesClientUtils.waitUntilMessagesAreReceived(tenant2ReceivingClient,
+                expectedMessageCount, runTime);
+
+        boolean tenant1SendSuccess = AndesClientUtils.getIfSenderIsSuccess(tenant1SendingClient, sendMessageCount1);
+        boolean tenant2SendSuccess = AndesClientUtils.getIfSenderIsSuccess(tenant2SendingClient, sendMessageCount2);
+
+        Assert.assertTrue(tenant1SendSuccess, "Sending failed for tenant 1 user 1.");
+        Assert.assertTrue(tenant2SendSuccess, "Sending failed for tenant 2 user 1.");
+        Assert.assertEquals(tenant2ReceivingClient.getReceivedqueueMessagecount(), sendMessageCount2, "Tenant 2 client received the message published to Tenant1");
+        Assert.assertEquals(tenant1ReceivingClient.getReceivedqueueMessagecount(), sendMessageCount1, "Tenant 1 client received the message published to Tenant2");
+    }
```

Comment:
This message will not always be true in a test failed scenario since there can be scenarios where less messages are received that expected. Instead can we print the received message count in the output and say message count is incorrect?